### DISCUSSION TITLE
Load Roses in MLife

### DISCUSSION THREAD

**Discussion by lt72884** [OpenFAST/openfast#1938] [Q&A]
Ok, so i have the basics done with MLife and OpenFAST. I saw the section in MLife called load roses. Following the user manual, i filled out that section. I run the MLife command to execute MLife. It completes with no errors, however, im not sure what the load roses did or if the even ran. Here is what my settings look like, i  am not familiar with it so im not sure the settings are correct. 

Please help me get load rosing to work:)

https://photos.app.goo.gl/n3hBgPoUyWJHtueu7

thanks

**Comment by jjonkman** [comment]
Dear @lt72884,

Your load rose settings look fine to me.  It looks like you are calculating load roses for the xb and yb shear forces for the root of blade 1.  Do you see MLife outputs associated with your rose name "RootFMxy1"?

Best regards,

**Comment by lt72884** [comment]
that seems to be the issue, in all the excell files MLife created, none of them show a name of "RootFMxy1" or a category of "Load Rose" 

Nothing looks different in the excell files compared to files with no load roses. 

What should the output in excell look like? 

thanks

**Reply by jjonkman** [reply]
Some of the MLife CertTest cases make use of load roses.  I would suggest using one of them for reference, e.g, Test09.

Best regards,

**Comment by lt72884** [comment]
ok, so i ran the cert 9 test and nothing in the output is showing any type of load roses. What am i looking for? 

thanks

**Reply by jjonkman** [reply]
In MLife Test09, I see load rose results for rose names "RootFxyc1" and "TwHt1MLxyt" for statistics, lifetime damage, and lifetime DELs.

Best regards,

**Comment by lt72884** [comment]
Could you please post a screen shot of it in the excel file because i do not see it at all. I want to compare please :)

**Reply by jjonkman** [reply]
The results are provided in the _CertTest\NREL_Results\Test09_ directory of the MLife archive.

**Comment by lt72884** [comment]
i looked through those last night but i do not see any title or name of "Rose Loads" or the like

the out put looks no different than when i had or had no rose loading. 

ill keep digging though.

`
![no load roses](https://github.com/OpenFAST/openfast/assets/52751207/3c9873b5-bc04-4160-8803-a4af974da367)
`

**Reply by jjonkman** [reply]
The load rose outputs show with _1, _2, etc. in their names.

**Comment by lt72884** [comment]
i thought the names such as _1, _2 where the names based on the channel names supplied to the settings file? IE in the screenshot attached, you have the channel name, the slope, LMF, LUlt etc. i thought the _1 etc were the names of the channels. 

Do i have to add RootFxb1_1 and RootFxb1_2 to the fatigue section? because my output doesnot have any of the underscored names at all

![calculated names](https://github.com/OpenFAST/openfast/assets/52751207/88ec2d38-56af-48cb-883e-6866b97e29de)

**Comment by lt72884** [comment]
here is my output with rose loading.. no underscores
![my output](https://github.com/OpenFAST/openfast/assets/52751207/45ec3069-2d74-4014-9185-8dabd204f8ce)

**Reply by jjonkman** [reply]
The channels in your MLife output do not seem to match the channels you are defining in the MLife input file; are you sure these outputs were generated by these inputs?

Are you able to run Test09 from the MLife archive and reproduce the NREL results for Test09?

**Comment by lt72884** [comment]
here are my actual inputs i used in my settings file that produced the above information based on the LUlt of 806kN

![inputs](https://github.com/OpenFAST/openfast/assets/52751207/ec052bec-4ce8-434d-9124-5b384d656425)

![load roses](https://github.com/OpenFAST/openfast/assets/52751207/a18a6d6a-a434-434c-8e8e-42291280cb52)


the other image (2 posts above with the long list), is of test09 settings file that has RootFxyc1_1 etc. 

i have completed the test09 and have the results as well. They match that of the NREL folder. 

for whatever reason, in my settings file (attached in this post along with rose settings), when i use the settings for load roses in my original post, nothing changes with the output. So i do not know what i am missing.

**Reply by jjonkman** [reply]
It looks like you've set "RootFMxy1" both as a calculated channel and as a load rose channel.  I'm not sure MLife would do in this case, but this use of the same variable to represent two different things could be problematic.  Do you get the load rose outputs you expect if you eliminate the Calculated Channels?

**Comment by lt72884** [comment]
that could be the issue. but right now, everything went up in smoke. For some reason, MATLAB out of nowhere is throwing a "error mkdir, device not ready"  or "error using + sign" i get these errors on multiple machines.. so i think i have to start all over again hahaha

**Comment by lt72884** [comment]
![rose load settings](https://github.com/OpenFAST/openfast/assets/52751207/002500ec-6081-4d30-8a72-8fa4c1139342)
here is the error im getting anytime i add the load rose settings in

![error from rose loads](https://github.com/OpenFAST/openfast/assets/52751207/ec3fb3a5-ca27-4ddd-9002-7626602cc643)

**Reply by jjonkman** [reply]
Sorry, but I'm not familiar with this error.  I'm a bit concerned that you have a space in your Col Title name ("RootForces x_y"); I'm not sure how MLife would interpret that.  If that doesn't solve the problem, you'll have to debug what MLife doesn't like within MATLAB.

**Comment by lt72884** [comment]
ok, i will look into that space.  Once i remove all the rose load settings, things work as normal. I will continue soon with load roses. 

whats the advantage to load roses? What is it actually doing?

thansk

**Reply by jjonkman** [reply]
A load rose considers loads at different orientations within a given cross section.  For fatigue, a load rose ensures that fatigue cycles are counted uniquely at different orientations within the cross section.  This is important because the maximum load may not occur along the primary axes of a cross section.

Best regards,

**Comment by lt72884** [comment]
after fixing the space, i still get the same error. I can not figure out what is going on. Its only when i add settings for the load roses. if i remove all my load rose settings, it runs just fine. 

the cert test runs just fine too. haha

do i need to add the rose name to the fatigue section and fill those columns out? 

IE, Channel# NSlope SNslope etc etc?

thanks
